pythnet/remote-executor/programs/remote-executor/src/state/governance_payload.rs

```diff
@@ -0,0 +1,125 @@
+    fn deserialize(buf: &mut &[u8]) -> std::result::Result<BigEndianU16, std::io::Error> {
+        if buf.len() < size_of::<u16>() {
+            return Err(std::io::Error::new(
+                ErrorKind::InvalidInput,
+                "Unexpected length of input",
+            ));
+        }
+        let res = u16::from_be_bytes(buf[..size_of::<u16>()].try_into().unwrap());
+        *buf = &buf[size_of::<u16>()..];
+        Ok(BigEndianU16{ value:res})
+    }
+}
+
+impl AnchorSerialize for BigEndianU16 {
+    fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        writer.write_all(&self.to_be_bytes())
+    }
+}
+
+impl Deref for BigEndianU16 {
+    type Target = u16;
+
+    fn deref(&self) -> &Self::Target {
+        &self.value
+    }
+}
+
+/// InstructionData wrapper. It can be removed once Borsh serialization for Instruction is supported in the SDK
+#[derive(Clone, Debug, PartialEq, Eq, AnchorDeserialize, AnchorSerialize)]
+pub struct InstructionData {
+    /// Pubkey of the instruction processor that executes this instruction
+    pub program_id: Pubkey,
+    /// Metadata for what accounts should be passed to the instruction processor
+    pub accounts: Vec<AccountMetaData>,
+    /// Opaque data passed to the instruction processor
+    pub data: Vec<u8>,
+}
+
+/// Account metadata used to define Instructions
+#[derive(Clone, Debug, PartialEq, Eq, AnchorDeserialize, AnchorSerialize)]
+pub struct AccountMetaData {
+    /// An account's public key
+    pub pubkey: Pubkey,
+    /// True if an Instruction requires a Transaction signature matching `pubkey`.
+    pub is_signer: bool,
+    /// True if the `pubkey` can be loaded as a read-write account.
+    pub is_writable: bool,
+}
+
+impl From<&InstructionData> for Instruction {
+    fn from(instruction: &InstructionData) -> Self {
+        Instruction {
+            program_id: instruction.program_id,
+            accounts: instruction
+                .accounts
+                .iter()
+                .map(|a| AccountMeta {
+                    pubkey: a.pubkey,
+                    is_signer: a.is_signer,
+                    is_writable: a.is_writable,
+                })
+                .collect(),
+            data: instruction.data.clone(),
+        }
+    }
+}
+
+impl ExecutorPayload {
+    const MODULE : Module = Module::Executor;
+    const ACTION : Action = Action::ExecutePostedVaa;
+
+    pub fn check_header(&self) -> Result<()>{
+        assert_or_err(self.header.magic_number == MAGIC_NUMBER, err!(ExecutorError::GovernanceHeaderInvalidMagicNumber))?;
+        assert_or_err(self.header.module == ExecutorPayload::MODULE, err!(ExecutorError::GovernanceHeaderInvalidMagicNumber))?;
+        assert_or_err(self.header.action == ExecutorPayload::ACTION, err!(ExecutorError::GovernanceHeaderInvalidMagicNumber))?;
+        assert_or_err(Chain::from(self.header.chain.value) == Chain::Pythnet, err!(ExecutorError::GovernanceHeaderInvalidMagicNumber))
```

Comment:
Have you seen the [boolinator](https://crates.io/crates/boolinator) crate? It includes a helpful [`as_result`](https://docs.rs/boolinator/2.4.0/boolinator/trait.Boolinator.html#tymethod.as_result) function that lets you do stuff like
```rust
some_cond.not().as_result((), Err(..))?;
```

which avoids the need for `assert_or_err`.

Reply:
https://github.com/pyth-network/pyth-crosschain/pull/288